File: src/interface/reservation.ts
import { ResponseUser } from "./auth";
import { Listing } from "./listing";

export interface Reservation {
    id: string;
    userId: string;
    listingId: string;
    startDate: Date;
    endDate: Date;
    totalPrice: number;
    createdAt: Date;
  
    user: ResponseUser; // Assuming User interface is defined
    listing: Listing; // Assuming Listing interface is defined
  }
  

File: backend/src/main/java/com/example/airbnb/document/Reservation.java
package com.example.airbnb.document;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "reservations")
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Setter
public class Reservation {
    @Id
    private String id;
    private String accommodationId;
    private Date startDate;
    private Date endDate;
    private String status;
    private int totalPrice;

    @DBRef
    private User user;
    @DBRef
    private Accommodation accommodation;
}
